crates/ika-core/src/dwallet_mpc/dwallet_mpc_service.rs

```diff
@@ -296,6 +304,73 @@ impl DWalletMPCService {
         }
     }
 
+    /// Receive all completed MPC sessions from the MPC Output Verifier over the `consensus_round_completed_sessions` channel.
+    /// If the session exists, mark is as `MPCSessionStatus::Finished`.
+    /// Otherwise, create a new session with that status, in order to avoid re-running the computation for it.
+    fn receive_completed_mpc_session_identifiers(&mut self, bootstrap: bool) {
+        let mut completed_sessions = HashSet::new();
+        loop {
+            match self.consensus_round_completed_sessions_receiver.try_recv() {
+                Err(mpsc::error::TryRecvError::Empty) => {
+                    // No more completed sessions to report at the moment.
+                    break;
+                }
+                Err(e) => {
+                    error!(
+                        authority=?self.epoch_store.name,
+                        e=?e,
+                        "error in reading completed session IDs"
+                    );
+
+                    break;
+                }
+                Ok(completed_session_identifier) => {
+                    debug!(
+                        validator=?self.epoch_store.name,
+                        completed_session_identifier=?completed_session_identifier,
+                        bootstrap=bootstrap,
+                        "Received completed session identifier"
+                    );
+                    // There might be more completed sessions to report, so report this one and continue receiving (don't break).
+                    completed_sessions.insert(completed_session_identifier);
+                }
+            }
+        }
+
+        debug!(
+            validator=?self.epoch_store.name,
+            completed_sessions=?completed_sessions,
+            bootstrap=bootstrap,
+            "Received completed session identifiers"
+        );
+
+        // self.last_read_consensus_round is the current reading consesnus round, also add self.last_db_consesnus_round that before it we don't compute.
+        // maybe we can get this last_db_consesnus_round from the dag from the db. highest_known_commit_at_startup maybe its not the consensus round tho
+        // read if the sui consensus syncs these values somehow
```

Comment:
Can you rephrase it? I can't understand any of it.

Reply:
sorry I should have deleted it

Reply:
deleted